tls/README.md

```diff
@@ -0,0 +1,287 @@
+              "ValidityDuration": "10y",
+              "Usages": [
+                "KeyUsageDigitalSignature",
+                "KeyUsageKeyEncipherment",
+                "KeyUsageCertSign"
+              ],
+              "ExtendedUsages": []
+            }
+          ]
+        },
+        "Status": {
+          "Errors": null
+        }
+...
+  CertKeyPairs": {
+    "Items": [
+      {
+        "LogicalName": "",
+        "Description": "",
+        "Name": "metrics.openshift-apiserver-operator.svc::3074380002740555304",
+        "Spec": {
+          "SecretLocations": [
+            {
+              "Namespace": "openshift-apiserver-operator",
+              "Name": "openshift-apiserver-operator-serving-cert"
+            }
+          ],
+          "OnDiskLocations": null,
+          "CertMetadata": {
+            "CertIdentifier": {
+              "CommonName": "metrics.openshift-apiserver-operator.svc",
+              "SerialNumber": "3074380002740555304",
+              "Issuer": {
+                "CommonName": "openshift-service-serving-signer@1701354509",
+                "SerialNumber": "",
+                "Issuer": null
+              }
+            },
+            "SignatureAlgorithm": "SHA256-RSA",
+            "PublicKeyAlgorithm": "RSA",
+            "PublicKeyBitSize": "2048 bit",
+            "ValidityDuration": "2y",
+            "Usages": [
+              "KeyUsageDigitalSignature",
+              "KeyUsageKeyEncipherment"
+            ],
+            "ExtendedUsages": [
+              "ExtKeyUsageServerAuth"
+            ]
+          },
+          "Details": {
+            "CertType": "ServingCertDetails",
+            "SignerDetails": null,
+            "ServingCertDetails": {
+              "DNSNames": [
+                "metrics.openshift-apiserver-operator.svc",
+                "metrics.openshift-apiserver-operator.svc.cluster.local"
+              ],
+              "IPAddresses": null
+            },
+            "ClientCertDetails": null
+          }
+        },
+        "Status": {
+          "Errors": null
+        }
+      },
+```
+
+This stores the following info:
+* all secrets / configmaps which are certificates along with their metadata
+* parses them and stores metadata
+* removes locations that are versioned or hashed copies of existing certificates
+* deduplicates them by content, so copies of certificates are grouped by location
+
+## Certificate metadata
+
+Certificate contents may however be insufficient - i.e. it's not clear which OpenShift is responsible 
+for the certificate lifecycle. To do that the certs should have additional annotations.
+
+Recently we've added the `openshift.io/owning-component: foobar` annotation to (almost) all certificates
+so that issues related to this certificate would be routed to a proper location. For instance, 
+problems with `service-ca` certificates are filed in Jira for "Networking / cluster-network-operator" component 
```

Comment:
The service-ca is owned by the service-ca component, not the networking component